test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs

```diff
@@ -49,99 +46,326 @@ public void ToOtlpResourceMetricsTest(bool includeServiceNameInResource)
 
-                Assert.Single(request.ResourceMetrics);
-                var resourceMetric = request.ResourceMetrics.First();
-                var oltpResource = resourceMetric.Resource;
+            if (attributes.Length > 0)
+            {
+                OtlpTestHelpers.AssertOtlpAttributes(attributes, dataPoint.Attributes);
+            }
+            else
+            {
+                Assert.Empty(dataPoint.Attributes);
+            }
 
-                if (includeServiceNameInResource)
-                {
-                    Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.StringValue == "service-name");
-                    Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceNamespace && kvp.Value.StringValue == "ns1");
-                }
-                else
-                {
-                    Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString().Contains("unknown_service:"));
-                }
+            Assert.Empty(dataPoint.Exemplars);
 
-                Assert.Single(resourceMetric.InstrumentationLibraryMetrics);
-                var instrumentationLibraryMetrics = resourceMetric.InstrumentationLibraryMetrics.First();
-                Assert.Equal(string.Empty, instrumentationLibraryMetrics.SchemaUrl);
-                Assert.Equal(meter.Name, instrumentationLibraryMetrics.InstrumentationLibrary.Name);
-                Assert.Equal("0.0.1", instrumentationLibraryMetrics.InstrumentationLibrary.Version);
+#pragma warning disable CS0612 // Type or member is obsolete
+            Assert.Null(actual.IntGauge);
+            Assert.Null(actual.IntSum);
+            Assert.Null(actual.IntHistogram);
+            Assert.Empty(dataPoint.Labels);
+#pragma warning restore CS0612 // Type or member is obsolete
+        }
 
-                Assert.Single(instrumentationLibraryMetrics.Metrics);
+        [Theory]
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, null, 123.45, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Delta, true)]
+        [InlineData("test_counter", "description", "unit", 123, null, AggregationTemporality.Cumulative, true)]
+        [InlineData("test_counter", null, null, 123, null, AggregationTemporality.Delta, true, "key1", "value1", "key2", 123)]
+        public void TestCounterToOltpMetric(string name, string description, string unit, long? longValue, double? doubleValue, AggregationTemporality aggregationTemporality, bool isMonotonic, params object[] keysValues)
+        {
+            var metrics = new List<Metric>();
+
+            var metricReader = new BaseExportingMetricReader(new InMemoryExporter<Metric>(metrics));
```

Comment:
Minor: consider this if we don't need the metricReader local variable https://github.com/open-telemetry/opentelemetry-dotnet/blob/fb161af9393f1eb2a6bae077f2478700a1623632/test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs#L38-L42